src/seedu/addressbook/data/person/ReadOnlyPerson.java

```diff
@@ -83,4 +84,12 @@ default String getAsTextHidePrivate() {
         }
         return builder.toString();
     }
+
+    default String getPrintableString(Printable... printables) {
+        StringBuilder builder = new StringBuilder();
```

Comment:
Good job using StringBuilder. You may also take a look at StringJoiner and String.Join